### DISCUSSION TITLE
Slash Commands and guild specific permissions ?

### DISCUSSION THREAD

**Discussion by nmarotte** [Pycord-Development/pycord#549] [Q&A] [answered]
Hi, I'm trying to build a bot with slash commands but I am not able to setup role/permission specific commands

For example I would like that my ban command only works for the member with a specific role, or the members with the ban permission (type=4)

However I think there is currently no way for that ? I've looked through all the examples of https://github.com/Pycord-Development/pycord/blob/master/examples/app_commands/slash_perms.py , but they require to hardcode the user_id for the command. There is no way to specify that the decorator must take the user_id of the person trying to invoke the command and checking if that person has a guild-specific role or that person has a certain permission

Now, if I want to make it role specific, it has to be hardcoded, since in the decorator there is no reference to self, the only value I can give for the `role_id` parameter is a hard coded integer/str. There is no way to setup a config/database for that value since there is no reference to the guild in which the command was invoked

And even if there was a way to reference the Cog in the permission decorator (or parameter of slash_command() ), it leaves us with only a reference to all the guilds the bot is in, and not just the guild in which the command is invoked, so it is also hard to retrieve the role_id specific to this guild unless if we hard code it

**Comment by BobDotCom** [comment, marked as answer]
Impossible at the current time due to discord-side limitations. When slash command permissions v2 releases, this should be possible.

**Comment by nmarotte** [comment]
I'm once again considering using pycord for my discord bot, has this feature been implemented yet ? Any way to follow the status of the slash command permissions v2 ?

Thanks ! 
@BobDotCom
